### PR TITLE
Atomised Radioactivity Rework from Nuclear Engine 

### PR DESCRIPTION
**Pull request by amylizzle**
## About the PR 
This is the radioactivity rework from my nuclear engine PR (#8767) split off for your reviewing pleasure.

Radiation design doc: https://hackmd.io/muJbjLJ7TIGKIaS6WdG1Og

## Why's this needed? 
Radiation as it currently stands is _bad_, in that it's basically an invisible kick in the teeth that kills you until you get a rad suit, and then you can ignore it. This sucks.

## Changelog

```changelog
(u)Amylizzle
(*)Changes how radiation works. All radiation sources are now visible, with a green or blue highlight and a glowing light!
(*)You now have an accumulated dose, which starts to make you sick over time. 
(*)Radiation protection now stacks and approaches 100% but never reaches it. Better not hang on to those radioactive ores!
(*)Radiation resistance is now split into two forms: Internal and external. Wearing a rad suit will no longer protect you from eating radium.
```

### CODE REVIEW

**Comment by amylizzle** [comment]
!merge_upstream

**Comment by ThatFiveGuys** [comment]
if not the case already, the tiles that have radiation on them should probably be visible too instead of (from what it sounds like) only the radioactive material producing light, that way you cant stack 20 erebite behind a wall made of copper and have people be irradiated from the hall by invisible to them radiation.

**Comment by ThatFiveGuys** [comment]
Wait doesnt radiation only applying to LoS mean you can just build a few walls around the reactor with a single false door entrance thats out of the way and then basically ignore the radioactive effects of a meltdown?

**Comment by ThatFiveGuys** [comment]
> There's a pretty clear glow from the light. Don't stand next to the glowy thing, in the glow. image

But for things with wider range then a tile it seems very visually misleading, its pretty fair for someone to expect that if theyre standing 5 tiles away from the nearest glowing light they wont get irradiated yet if the thing is radioactive enough they could, having tiles with radiation on them glow would also allow you to remove the LoS aspects. 

> It does not, because the reactor shoots out neutrons which penetrate walls based on their density and induce radioactivity in things they hit. Also it explodes.
Why doesn't the nuclear reactor use the radiation system?? It shooting out projectiles that are completely unrelated to all other radiation seems really weird.

**Comment by amylizzle** [comment]
discussed on discord, gonna stick with how it works now.

File: code/obj/machinery/porters.dm [hunk 1]
```diff
@@ -784,15 +784,15 @@ var/global/list/portable_machinery = list() // stop looping through world for th
 
 					if(51 to 70) //A nice tan
 						for(var/mob/living/carbon/M in src.contents)
-							M.changeStatus("radiation", 20 SECONDS, 1)
+							M.take_radiation_dose(0.5)
```
Comment (ZeWaka) [inline code/obj/machinery/porters.dm:787]: 0.5 WHATS

**Review by ZeWaka — CHANGES_REQUESTED** [review]

File: code/obj/machinery/porters.dm (hunk 1 above)
Comment (amylizzle) [reply code/obj/machinery/porters.dm]: SIEVERTS

File: code/modules/status_system/statusEffects.dm [hunk 2]
```diff
@@ -345,209 +345,111 @@
...
+				if(1 to 3)
+					. = 2 //you're getting into dangerous teritory
+				if(3 to 5)
+					. = 3 //you're at a 50/50 of kicking it
+				if(5 to 7)
+					. = 4 //more like 70/30 now
+				if(7 to 10)
+					. = 5 //you will die without treatment
+				if(7 to INFINITY)
+					. = 6 //you will die.
```
Comment (Tarmunora) [inline code/modules/status_system/statusEffects.dm:397]: imo should pull the `/0.4` factor down into the switch branches for readability - dividing by `0.4` makes it annoying to mentally translate rad dose into rad stage

File: code/modules/status_system/statusEffects.dm [hunk 3]
```diff
@@ -345,209 +345,111 @@
...
+				if(!ON_COOLDOWN(M, "radiation_stun_check", 1 SECOND) && prob((stage-1)**2) && M.bioHolder && !M.bioHolder.HasEffect("revenant"))
+					M.changeStatus("weakened", 3 SECONDS)
 					boutput(M, "<span class='alert'>You feel weak.</span>")
 					M.emote("collapse")
-			update_lights()
-			icon_state = "nradiation[stage]"
+				if(!ON_COOLDOWN(M, "radiation_vomit_check", 5 SECOND) && prob(stage**2) && M.bioHolder && !M.bioHolder.HasEffect("revenant"))
+					M.changeStatus("weakened", 3 SECONDS)
+					boutput(M, "<span class='alert'>You feel sick.</span>")
+					M.vomit()
```
Comment (Tarmunora) [inline code/modules/status_system/statusEffects.dm:452]: gut feel is that this is probably too much stunning, especially at higher stages, but can't say with certainty

File: code/obj/decal/cleanable.dm [hunk 4]
```diff
@@ -1721,14 +1721,14 @@ var/list/blood_decal_violent_icon_states = list("floor1", "floor2", "floor3", "f
...
-/obj/decal/cleanable/proc/streak_cleanable(var/list/directions, var/randcolor = 0, var/full_streak)
+/obj/decal/cleanable/proc/streak_cleanable(var/list/directions, var/randcolor = 0, var/full_streak, var/dist_lower=1, var/dist_upper=6)
 	if(src.disposed)
 		return
 	if(isnull(get_turf(src)))
 		CRASH("Attempting to streak cleanable [src] which is in null.")
 
 	var/destination
-	var/dist = rand(1,6)
+	var/dist = rand(dist_lower,dist_upper)
```
Comment (Tarmunora) [inline code/obj/decal/cleanable.dm:1731]: seems unrelated?

File: code/obj/item.dm [hunk 5]
```diff
@@ -435,7 +435,7 @@
 		SPAWN(1 SECOND)
 			if (!src || !M || !user)
 				return
-			SEND_SIGNAL(M, COMSIG_ITEM_CONSUMED, user, src)
+			SEND_SIGNAL(src, COMSIG_ITEM_CONSUMED, user, src)
```
Comment (Tarmunora) [inline code/obj/item.dm:438]: additionally, here, `user` is forcefeeding `M` a thing... and the signal args do not reference `M` at all

**Review by Tarmunora — CHANGES_REQUESTED** [review]
Additionally, I think that antirad medicines could either use a buff, or there medbay should get some other of clearing high rad doses in a timely matter. A single radbow shot (assuming no radprot etc etc) takes over 2 minutes, and approx 30u of pentetic acid to fully clear up (unless my math is off).

File: code/obj/decal/cleanable.dm (hunk 4 above)
Comment (amylizzle) [reply code/obj/decal/cleanable.dm:1731]: oh whoops, yeah, that's for the nuke engine - I'll remove it

File: code/obj/item.dm [hunk 6]
```diff
@@ -401,7 +401,7 @@
 		SPAWN(0.6 SECOND)
 			if (!src || !M || !user)
 				return
-			SEND_SIGNAL(M, COMSIG_ITEM_CONSUMED, user, src)
+			SEND_SIGNAL(src, COMSIG_ITEM_CONSUMED, user, src)
```
[the remark this answers is not in the document]
Comment (amylizzle) [reply code/obj/item.dm:404]: You are right, there's only one other thing that uses this signal and I misread it. However, I think the proper fix would be to create a new signal of `COMSIG_MOB_ITEM_CONSUMED` and have the other thing trigger on that, since the rest of the `COMSIG_ITEM` signals trigger on the item, and `COMSIG_MOB` triggers on the mob.

File: code/modules/status_system/statusEffects.dm (hunk 3 above)
Comment (amylizzle) [reply code/modules/status_system/statusEffects.dm:452]: It's pretty variable because of the `prob()` but at low stages it's pretty rare, at mid stages it's an irritation, and at end stages it's every few seconds, but at that point you're already well into dying. I can lower it if you prefer, but it feels about right to me.

File: code/obj/item.dm (hunk 6 above)
[the remark this answers is not in the document]
Comment (amylizzle) [reply code/obj/item.dm:404]: In fact, all of the `COMSIG_ITEM_CONSUMED` signals seem wrong, they all target mobs instead of items despite being named like other item signals

[the remark this answers is not in the document]
Comment (amylizzle) [reply code/obj/item.dm]: Fixing this in another PR

[the remark this answers is not in the document]
Comment (amylizzle) [reply code/obj/item.dm]: #9763

File: code/modules/chemistry/Reagents-Medical.dm [hunk 7]
```diff
@@ -605,13 +605,12 @@ datum
 			transparency = 40
 			value = 2 // 1c + 1c
 			target_organs = list("left_kidney", "right_kidney", "liver")
+			threshold = 5
```
Comment (Tarmunora) [inline code/modules/chemistry/Reagents-Medical.dm:608]:
```suggestion
			threshold = THRESHOLD_INIT
			threshold_volume = 5
```

**Comment by Flaborized** [comment]
Is there a place where the precise changes are documented? I read the design doc (and liked and agreed with what's in there) but it's mostly like, design rules instead of a description of the changes without like, reading through and understanding the code. I'm assuming this exists somewhere and I just missed it, though!

Main questions I have are like: How does this affect current implementations of radiation? Are radbows, gene scanner scramblers, chems, radstorms, etc, all intended to be relatively the same level of dangerous? Is the accumulated dose mechanic shown to players in stuff like health scanners, if so is it specific like old radiation (ie: a unit of measurement and a number) or is it like current where it tells you "stage 1" or "stage 2"? If you can't have 100% protection anymore, *how* much protection is what would have been 100%? Is it like, functional immunity but not if you hold cerenkite for 20 minutes or is it still a pretty serious threat with relatively short exposure?

I'm mostly writing all these questions because I'm interested in all this, the design doc is really good in my eyes and making radiation a more complicated, involved system is very cool and good. I think some kind of brief overview of these specifics in the PR body (or a link to something like that) would be very helpful for people who wanna know more about the cool ideas and features but are too lazy to read the code. If I missed something like this here please let me know and ignore me :shelterfrog:

**Comment by amylizzle** [comment]
These are good questions. I shall do my best to answer them.
>Is there a place where the precise changes are documented? 

No, not really. Sorry. The changelog covers the broad strokes, the code covers the finer points. To my knowledge, existing radiation isn't well documented either, just a dead link on the wiki and a few mentions on doctoring page.

>How does this affect current implementations of radiation? Are radbows, gene scanner scramblers, chems, radstorms, etc, all intended to be relatively the same level of dangerous? 

I've tried for parity where possible. If something is substantially different, that's a mistake on my part. It wasn't possible to achieve perfect parity, because the systems are quite different though. Radbows were explicitly tested, and now it's basically "one shot for slow death, two for quick death" with one-shot being manageable with meds.

>Is the accumulated dose mechanic shown to players in stuff like health scanners, if so is it specific like old radiation (ie: a unit of measurement and a number) 

it is! Now radiation is show in a medscanner as a level and also a specific dose (in Sieverts). The level gives doctors a rough guide of how screwed the patient is, and the number allows precise monitoring for treatment effectiveness etc.

>If you can't have 100% protection anymore, how much protection is what would have been 100%? Is it like, functional immunity but not if you hold cerenkite for 20 minutes or is it still a pretty serious threat with relatively short exposure?

So, a full radsuit which would previously give you immunity now gives you 99% protection. It is as you say, functional immunity so long as you don't do something very stupid like go swimming in radium.

File: code/modules/medical/pathology/pathogen_symptoms.dm [hunk 8]
```diff
@@ -2013,24 +2013,24 @@ datum/pathogeneffects/malevolent/radiation
...
-					M.changeStatus("radiation", rand(2,4) SECONDS)
+					M.take_radiation_dose( rand(2,4) SECONDS)
 					boutput(M,"<span class='alert'>You feel extremely sick!!</span>")
 				else if (prob(20))
-					M.changeStatus("radiation", 3 SECONDS)
+					M.take_radiation_dose( 3 SECONDS)
 					boutput(M,"<span class='alert'>You feel very sick!</span>")
 				else if (prob(40))
-					M.changeStatus("radiation", 2 SECONDS)
+					M.take_radiation_dose( 2 SECONDS)
```
Comment (ZeWaka) [inline code/modules/medical/pathology/pathogen_symptoms.dm:2033]: weird spacing

File: code/modules/medical/genetics/geneticsMachines.dm [hunk 9]
```diff
@@ -561,7 +561,7 @@
 			if (addEffect) // re-mutantify if we would have been able to anyway
 				subject.bioHolder.AddEffect(addEffect)
 			if (genResearch.emitter_radiation > 0)
-				subject.changeStatus("radiation", (genResearch.emitter_radiation) SECONDS, 3)
+				subject.take_radiation_dose( 0.5 SIEVERTS)
```
Comment (ZeWaka) [inline code/modules/medical/genetics/geneticsMachines.dm:564]:
```suggestion
				subject.take_radiation_dose(0.5 SIEVERTS)
```

File: code/modules/medical/genetics/bioEffects/harmful.dm [hunk 10]
```diff
@@ -400,17 +400,13 @@
 		if (ishuman(owner))
 			overlay_image = image("icon" = 'icons/effects/genetics.dmi', "icon_state" = "aurapulse", layer = MOB_LIMB_LAYER)
 			overlay_image.color = "#BBD90F"
+		owner.AddComponent(/datum/component/radioactive,50,FALSE,FALSE)
```
Comment (ZeWaka) [inline code/modules/medical/genetics/bioEffects/harmful.dm:403]: spacing
```suggestion
		owner.AddComponent(/datum/component/radioactive, 50, FALSE, FALSE)
```

File: code/modules/materials/Mat_ProcsDefines.dm [hunk 11]
```diff
@@ -189,6 +189,8 @@ var/global/list/triggerVars = list("triggersOnBullet", "triggersOnEat", "trigger
 	if(appearance)
 		src.setMaterialAppearance(mat1)
 
+	if(src.material)
```
Comment (ZeWaka) [inline code/modules/materials/Mat_ProcsDefines.dm:192]: ?.

File: code/modules/materials/Mat_MaterialProcs.dm [hunk 12]
```diff
@@ -446,37 +426,29 @@ triggerOnEntered(var/atom/owner, var/atom/entering)
...
+		location.AddComponent(/datum/component/radioactive, location.material.getProperty("n_radioactive")*10, FALSE,TRUE)
 		return
 
-/datum/materialProc/n_radioactive_pickup
-	execute(var/mob/M, var/obj/item/I)
-		if(I.material)
-			M.changeStatus("n_radiation", I.material.getProperty("n_radioactive") * 2 SECONDS)
+/datum/materialProc/n_radioactive_remove
+	execute(var/atom/location)
+		animate_flash_color_fill_inherit(location,"#1122EE",-1,40)
```
Comment (ZeWaka) [inline code/modules/materials/Mat_MaterialProcs.dm:449]: spacing for here and friends above

File: code/modules/hydroponics/plants_weed.dm [hunk 13]
```diff
@@ -158,7 +158,7 @@ ABSTRACT_TYPE(/datum/plant/weed)
 					radstrength = 50
 					radrange = 3
 			for (var/mob/living/carbon/M in range(radrange,POT))
-				M.changeStatus("radiation", (radstrength) SECONDS, 3)
+				M.take_radiation_dose( radstrength/10 SIEVERTS)
```
Comment (ZeWaka) [inline code/modules/hydroponics/plants_weed.dm:161]: 😨 and more everywhere
```suggestion
				M.take_radiation_dose(radstrength/10 SIEVERTS)
```

**Comment by amylizzle** [comment]
This is what the radiation resistance curve looks like now. Purple is old style, black is new.
image

File: _std/math.dm [hunk 14]
```diff
@@ -92,3 +92,6 @@ proc/text2num_safe(x)
 /proc/fixed_random(x, y)
 	. = sin(x * 12.9898 + y * 78.233) * 43758.5453
 	. -= round(.)
+
+// hyperbolic trig functions
+#define tanh(x) (((eulers**(2*x))-1)/((eulers**(2*x))+1))
```
Comment (ZeWaka) [inline _std/math.dm:97]: \>not the C impl

File: code/datums/components/radioactive.dm [hunk 15]
```diff
@@ -0,0 +1,145 @@
...
+
+		if(isitem(parent))
+			RegisterSignal(parent, list(COMSIG_ITEM_PROCESS), .proc/ticked)
+			if(!(parent in global.processing_items))
+				global.processing_items.Add(parent)
+				src._added_to_items_processing = TRUE
+		else
+			global.processing_items.Add(src) //gross - in the event that this component is put on something that isn't an item, use the item processing loop anyway
+		var/atom/PA = parent
+		var/color = (neutron ? "#2e3ae4" : "#18e022") + num2hex(min(128,round(255*radStrength/100)),2) //base color + alpha
```
Comment (ZeWaka) [inline code/datums/components/radioactive.dm:54]: Spacing here and elsewhere

File: code/mob/living/critter/small_animal.dm [hunk 16]
```diff
@@ -116,6 +116,7 @@ ABSTRACT_TYPE(/mob/living/critter/small_animal)
 		remove_lifeprocess(/datum/lifeprocess/sight)
 		remove_lifeprocess(/datum/lifeprocess/skin)
 		remove_lifeprocess(/datum/lifeprocess/statusupdate)
+		remove_lifeprocess(/datum/lifeprocess/radiation)
```
Comment (ZeWaka) [inline code/mob/living/critter/small_animal.dm:119]: Why

Comment (amylizzle) [reply code/mob/living/critter/small_animal.dm:119]: No point ticking the lifeprocess after the critter is dead?

File: _std/math.dm (hunk 14 above)
Comment (amylizzle) [reply _std/math.dm:97]: how do?

**Comment by amylizzle** [comment]
Calculator showing how damage is applied based on dosage, along with how that corresponds to seconds standing next to a bar of erebite. Vary the rad resistance (t) value to see how that works out.

https://www.desmos.com/calculator/ijdtyvyknl

**Comment by Tarmunora** [comment]
I'm going to be unable to continue review for a week or so, DNM until then.

File: code/mob/living/life/radiation.dm [hunk 17]
```diff
@@ -0,0 +1,13 @@
+/datum/lifeprocess/radiation
+	process(datum/gas_mixture/environment)
+		if(!owner || !owner.radiation_dose)
+			return
+
+		//apply effects
+		if(!owner.hasStatus("radiation"))
+			owner.changeStatus("radiation",null)
```
Comment (ZeWaka) [inline code/mob/living/life/radiation.dm:8]: this is confusing, what is the purpose of this

File: code/datums/components/radioactive.dm [hunk 18]
```diff
@@ -0,0 +1,145 @@
...
+	/// Does this radiation source slowly decay over time? If so, it will take ~ 6 * radStrength seconds to decay completely.
+	var/decays = FALSE
+	/// Is this radiation source a neutron source? If it is, it does more damage per dose. Associated with n_radiation mat property.
+	var/neutron = FALSE
+	/// Internal, do not touch - keeps a record of whether or not we had to add this to the item processing list
+	var/_added_to_items_processing = FALSE
+	/// How wide a range this radiation source affects. Greater than one should be very rarely used, since all atoms in this range will be exposed per tick
+	var/effect_range = 1
+	/// Internal, do not touch - keeps a record of atom.color since we override it with filters.
+	var/backup_color = null //so hacky
```
Comment (ZeWaka) [inline code/datums/components/radioactive.dm:23]: consider `_backup_color` then

File: code/mob/living/life/radiation.dm (hunk 17 above)
Comment (amylizzle) [reply code/mob/living/life/radiation.dm:8]: I'll add code comments to clear this up, but essentially it just gives the owner an infinite duration radiation status effect, assuming their dose is greater than 0 (early return checks for this above)

File: code/mob/living/life/radiation.dm [hunk 19]
```diff
@@ -0,0 +1,16 @@
...
+
+		//apply effects - by this point you have received a non-zero dose, so give the user an infinite duration radiation effect
+		//if they haven't got one already
+		if(!owner.hasStatus("radiation"))
+			owner.changeStatus("radiation",null)
+
+		if(!isdead(owner))
+			//remove some rads
+			tick_count++
+			boutput(world, "T: [tick_count] R: [owner.radiation_dose] D: [(src.get_multiplier() * (owner.radiation_dose_decay * (owner.radiation_dose**1.3))/2)]")
```
Comment (Tarmunora) [inline code/mob/living/life/radiation.dm:15]:
```suggestion
```

**Comment by amylizzle** [comment]
Falloff and damage is this now, with the last 0.35Sv taking a while to decay entirely - so don't get repeatedly dosed, it'll have consequences.
image

**Comment by amylizzle** [comment]
Yeah, I'm pretty happy with where this is right now. I'm callin' it done, barring any minor balance tweaks that are suggested. @Tarmunora plz judge.

File: code/modules/chemistry/Reagents-PoisonEtc.dm [hunk 20]
```diff
@@ -1415,7 +1415,7 @@ datum
 
 			on_mob_life(var/mob/M, var/mult = 1)
 				if (!M) M = holder.my_atom
-				M.changeStatus("radiation", 2 SECONDS * mult)
+				M.take_radiation_dose(0.1 SIEVERTS * mult, internal=TRUE)
```
Comment (Tarmunora) [inline code/modules/chemistry/Reagents-PoisonEtc.dm:1418]:
```suggestion
				M.take_radiation_dose(0.02 SIEVERTS * mult, internal=TRUE)
```
should match mutagen in rad intensity. Could maybe go to `0.03`, but old slurry was less than mutagen in the first place so 🤷

File: code/modules/events/radiation.dm [hunk 21]
```diff
@@ -71,7 +71,7 @@
 				return
 		animate_flash_color_fill_inherit(T,"#00FF00",1,5)
 		for (var/mob/living/carbon/M in T.contents)
-			M.changeStatus("radiation", (rad_strength) SECONDS, 3)
+			M.take_radiation_dose(rad_strength/10 SIEVERTS)
```
Comment (Tarmunora) [inline code/modules/events/radiation.dm:74]: this is a lethal dose

File: code/modules/events/radiation.dm [hunk 22]
```diff
@@ -122,8 +122,8 @@
 				animate_flash_color_fill_inherit(T,"#FFDD00",1,5)
 				return
 		animate_flash_color_fill_inherit(T,"#0084ff",1,5)
-		for (var/atom/A in T.contents)
-			A.changeStatus("n_radiation", (rad_strength) SECONDS, 3)
+		for (var/mob/A in T.contents)
+			A.take_radiation_dose(rad_strength/5 SIEVERTS)
```
Comment (Tarmunora) [inline code/modules/events/radiation.dm:126]: this ends up being weaker than the non-neutron one with current numbers (2Sv vs 2.5Sv).
Could make it actually irradiate stuff (w/ component etc) in the vicinity in the future?

File: code/modules/hydroponics/plants_weed.dm [hunk 23]
```diff
@@ -158,7 +158,7 @@ ABSTRACT_TYPE(/datum/plant/weed)
 					radstrength = 50
 					radrange = 3
 			for (var/mob/living/carbon/M in range(radrange,POT))
-				M.changeStatus("radiation", (radstrength) SECONDS, 3)
+				M.take_radiation_dose(radstrength/30 SIEVERTS)
```
Comment (Tarmunora) [inline code/modules/hydroponics/plants_weed.dm:161]: oh jeez these were pretty crazy even before, weren't they.
A single dose even at max potency wont' be lethal, but any more than that and people are going to have a bad time
Might want to cheat a bit and put a cooldown on the victim so they can't be slammed by a dozen stacked plants perhaps?

File: code/modules/hydroponics/plants_weed.dm [hunk 24]
```diff
@@ -158,7 +158,7 @@ ABSTRACT_TYPE(/datum/plant/weed)
 					radstrength = 50
 					radrange = 3
 			for (var/mob/living/carbon/M in range(radrange,POT))
```
Comment (Tarmunora) [inline code/modules/hydroponics/plants_weed.dm:160]:
```suggestion
			for (var/mob/living/carbon/M in view(radrange,POT))
```
lol lets not have these go through walls anymore

File: code/modules/medical/genetics/geneticsMachines.dm [hunk 25]
```diff
@@ -561,7 +561,7 @@
 			if (addEffect) // re-mutantify if we would have been able to anyway
 				subject.bioHolder.AddEffect(addEffect)
 			if (genResearch.emitter_radiation > 0)
-				subject.changeStatus("radiation", (genResearch.emitter_radiation) SECONDS, 3)
+				subject.take_radiation_dose(0.5 SIEVERTS)
```
Comment (Tarmunora) [inline code/modules/medical/genetics/geneticsMachines.dm:564]: this should take `genResearch.emitter_radiation` into account (probably have to tweak the numbers on that var to make sense with new rads), since that's how the "radiation dampeners" upgrades do

File: code/modules/medical/genetics/geneticsMachines.dm [hunk 26]
```diff
@@ -580,7 +580,7 @@
 				return
 			src.log_me(subject, "gene scrambled", E)
 			if (genResearch.emitter_radiation > 0)
-				subject.changeStatus("radiation", (genResearch.emitter_radiation) SECONDS, 3)
+				subject.take_radiation_dose(0.1 SIEVERTS)
```
Comment (Tarmunora) [inline code/modules/medical/genetics/geneticsMachines.dm:583]: same here

File: code/modules/medical/pathology/pathogen_symptoms.dm [hunk 27]
```diff
@@ -2013,24 +2013,24 @@ datum/pathogeneffects/malevolent/radiation
...
+					M.take_radiation_dose(rand(2,4) SECONDS)
 					boutput(M,"<span class='alert'>You feel extremely sick!!</span>")
 				else if (prob(20))
-					M.changeStatus("radiation", 3 SECONDS)
+					M.take_radiation_dose(3 SECONDS)
 					boutput(M,"<span class='alert'>You feel very sick!</span>")
 				else if (prob(40))
-					M.changeStatus("radiation", 2 SECONDS)
+					M.take_radiation_dose(2 SECONDS)
 					boutput(M,"<span class='alert'>You feel sick.</span>")
```
Comment (Tarmunora) [inline code/modules/medical/pathology/pathogen_symptoms.dm:2034]: on the one hand, patho is disabled, and probably forever, so this doesn't really matter
on the other hand, dear lord.

File: code/modules/mining/tile_events.dm [hunk 28]
```diff
@@ -141,10 +141,10 @@
...
 		for (var/mob/living/L in range(1,AST))
-			L.changeStatus("radiation", 5 SECONDS, 2)
+			L.take_radiation_dose(1.0 SIEVERTS)
 
 	onExcavate(var/turf/simulated/wall/auto/asteroid/AST)
 		if (..())
 			return
 		for (var/mob/living/L in range(1,AST))
-			L.changeStatus("radiation", 10 SECONDS, 2)
+			L.take_radiation_dose(2.0 SIEVERTS)
```
Comment (Tarmunora) [inline code/modules/mining/tile_events.dm:150]: 😰

File: code/modules/mining/ore.dm [hunk 29]
```diff
@@ -159,13 +159,13 @@
...
 		for (var/mob/living/L in range(1,AST))
-			L.changeStatus("radiation", 5 SECONDS, 2)
+			L.take_radiation_dose(0.5 SIEVERTS)
 
 	onExcavate(var/turf/simulated/wall/auto/asteroid/AST)
 		if (..())
 			return
 		for (var/mob/living/L in range(1,AST))
-			L.changeStatus("radiation", 10 SECONDS, 2)
+			L.take_radiation_dose(1.0 SIEVERTS)
```
Comment (Tarmunora) [inline code/modules/mining/ore.dm:168]: in a full radsuit, this works out to be about 0.1Sv per rock mined, not including anything from the ores on the floor
this will probably work up to dangerous levels pretty quickly when mining cerenkite in bulk, even with "100%" protection.
maybe tone this down

File: code/modules/status_system/statusEffects.dm [hunk 30]
```diff
@@ -345,209 +345,103 @@
...
-			if(M)
-				SEND_SIGNAL(M, COMSIG_MOB_GEIGER_TICK, stage)
-				if(locate(/obj/item/implant/health) in M)
-					src.visible = 1
-				else
-					src.visible = 0
-				if (prob(stage - !!(M.traitHolder?.hasTrait("stablegenes"))) && (M.bioHolder && !M.bioHolder.HasEffect("revenant")))
+
+			if(stage > 0 && !isdead(M))
+				if (prob(max(stage-(2+M.traitHolder?.hasTrait("stablegenes")),0)**2) && (M.bioHolder && !M.bioHolder.HasEffect("revenant")))
```
Comment (Tarmunora) [inline code/modules/status_system/statusEffects.dm:434]: should have a cooldown on the check here too probably

Comment (Tarmunora) [reply code/modules/status_system/statusEffects.dm:434]: and maybe actually bump the stage offset down by 1. will probably need some tweaking to get this feeling right

File: code/modules/transport/pods/warp_travel.dm [hunk 31]
```diff
@@ -151,7 +151,7 @@
 			logTheThing(LOG_COMBAT, T, "entered [src] at [log_loc(src)] and gibbed")
 			return
 		else
-			T.changeStatus("radiation", rand(5,25) SECONDS, 2)
+			T.take_radiation_dose(rand()*4 SIEVERTS)
```
Comment (Tarmunora) [inline code/modules/transport/pods/warp_travel.dm:154]: 😰

File: code/obj/item/mine.dm [hunk 32]
```diff
@@ -201,7 +201,7 @@
 		if (length(mobs))
 			for (var/mob/living/L in mobs)
 				if (istype(L))
-					L.changeStatus("radiation", 60 SECONDS)
+					L.take_radiation_dose(5 SIEVERTS)
```
Comment (Tarmunora) [inline code/obj/item/mine.dm:204]: 😰 😰 😰 😰 😰

Comment (Tarmunora) [reply code/obj/item/mine.dm:204]: this went from "take ~70 damage over 60 seconds" to "you die before 60 seconds"

File: code/obj/machinery/singularity.dm [hunk 33]
```diff
@@ -381,7 +381,7 @@ for some reason I brought it back and tried to clean it up a bit and I regret ev
 			if (H.wear_suit)
 				return
 		M.take_toxin_damage(12)
-		M.changeStatus("radiation", 4*(radius+1) SECONDS)
+		M.take_radiation_dose(0.2 SIEVERTS*(radius+1))
```
Comment (Tarmunora) [inline code/obj/machinery/singularity.dm:384]: this should... probably be clamped, just in case someone fixes this proc to actually work

File: code/procs/scanprocs.dm [hunk 34]
```diff
@@ -198,11 +198,9 @@
 
 
 	var/datum/statusEffect/simpledot/radiation/R = M.hasStatus("radiation")
-	var/datum/statusEffect/simpledot/radiation/NR = M.hasStatus("n_radiation")
-	if (R)
-		rad_data = "&emsp;<span class='alert'>Radiation poisoning: Lv [R.stage]</span>"
-	if (NR)
-		nrad_data = "&emsp;<span class='alert'>Neutron Radiation poisoning: Lv [NR.stage]</span>"
+	if (R?.stage)
+		rad_data = "&emsp;<span class='alert'>Radiation poisoning: Lv [R.stage] Dose: [M.radiation_dose] Sv</span>"
```
Comment (Tarmunora) [inline code/procs/scanprocs.dm:202]: probably don't need to include stage, since that only affects stun/mutation chances

**Review by Tarmunora — CHANGES_REQUESTED** [review]
numbers

File: code/modules/medical/pathology/pathogen_symptoms.dm (hunk 27 above)
Comment (amylizzle) [reply code/modules/medical/pathology/pathogen_symptoms.dm:2034]: I'mma leave it in the hope that someone tries out patho one day and gets massively irradiated for their troubles :satanfrog:

File: code/obj/item/mine.dm (hunk 32 above)
Comment (amylizzle) [reply code/obj/item/mine.dm:204]: I'm nerfing it from crazy, but I'm still going to make it a potentially lethal dose because dammit radiation landmines should be harmful!

File: code/procs/scanprocs.dm (hunk 34 above)
Comment (amylizzle) [reply code/procs/scanprocs.dm:202]: gonna make this print the same stuff as mousing over the status effect. I want docs to be able to figure out how fucked the patient is without necessarily needing to have an intuitive grasp of Sieverts.

**Review by Tarmunora — APPROVED** [review]
Would probably still make some tweaks after merge, but I think this is at least ready to be tested on live
